module/version_reconfigure_test.go

```diff
@@ -0,0 +1,260 @@
+package module_test
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.viam.com/test"
+	goutils "go.viam.com/utils"
+	"go.viam.com/utils/pexec"
+	"go.viam.com/utils/rpc"
+
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/robot/client"
+	"go.viam.com/rdk/testutils"
+	"go.viam.com/rdk/utils"
+)
+
+// TODO(pre-merge) refactor this somewhat-shared function into a test_helpers esque file.
+func connect(port string, logger golog.Logger) (robot.Robot, error) {
+	connectCtx, cancelConn := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancelConn()
+	for {
```

Comment:
[q] Why is there retry logic here? Can you just use `client.New`?

Reply:
We need this because this connecting can start happening before the viam server is up and listening on a port. I've documented the observed behavior that justifies the loop.

Reply:
That said, this goes away when rewriting the tests as you recommended.